core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -130,15 +130,17 @@ private[spark] class TaskSchedulerImpl private[scheduler](
 
   val mapOutputTracker = SparkEnv.get.mapOutputTracker
 
-  var schedulableBuilder: SchedulableBuilder = null
+  private val SCHEDULER_MODE_PROPERTY = "spark.scheduler.mode"
+  private var schedulableBuilder: SchedulableBuilder = null
   var rootPool: Pool = null
   // default scheduler is FIFO
-  private val schedulingModeConf = conf.get("spark.scheduler.mode", "FIFO")
+  private val schedulingModeConf = conf.get(SCHEDULER_MODE_PROPERTY, SchedulingMode.FIFO.toString)
   val schedulingMode: SchedulingMode = try {
     SchedulingMode.withName(schedulingModeConf.toUpperCase)
   } catch {
     case e: java.util.NoSuchElementException =>
-      throw new SparkException(s"Unrecognized spark.scheduler.mode: $schedulingModeConf")
+      throw new SparkException(s"Unrecognized $SCHEDULER_MODE_PROPERTY: $schedulingModeConf. " +
+        s"Supported modes: ${SchedulingMode.FAIR} or ${SchedulingMode.FIFO}.")
```

Comment:
`SchedulingMode` possible values are `FIFO`, `FAIR` and `NONE` but `NONE` is _unsupported_ value. I agree to support for potential values and it can be achieved by adding following logic to `SchedulingMode` object and can be used required places(2 times here and 1 time at `Pool`)
`def getSupportedValuesAsString(): String = values.filter(_ != NONE).mkString(", ")
SchedulingMode.getSupportedValuesAsString() // returns FIFO, FAIR`
WDYT?